src/tools/compiletest/src/runtest.rs

```diff
@@ -2247,13 +2260,16 @@ impl<'test> TestCx<'test> {
     // codegen tests (using FileCheck)
 
     fn compile_test_and_save_ir(&self) -> ProcRes {
-        let aux_dir = self.aux_output_dir_name();
-
         let output_file = TargetLocation::ThisDirectory(self.output_base_dir());
         let input_file = &self.testpaths.file;
-        let mut rustc =
-            self.make_compile_args(input_file, output_file, EmitMetadata::No, AllowUnused::No);
-        rustc.arg("-L").arg(aux_dir).arg("--emit=llvm-ir");
+        let mut rustc = self.make_compile_args(
+            input_file,
+            output_file,
+            EmitMetadata::No,
+            AllowUnused::No,
+            LinkToAux::Yes,
+        );
+        rustc.arg("--emit=llvm-ir");
```

Comment:
this will have the same problem as `-L aux_dir`, right? should we fix that here too?

Reply:
Hmm, there are multiple places around that add flags. If we wanted to remove all the chances of this happening in the future, we should probably have a

```
prepare_and_run_rustc(..., |cmd| { cmd.arg("--emit=llvm-ir") });
```

Would that be something you'd like?

Reply:
hmm, that sounds like a lot of changes to use the new API - can we add a `src/test/ui` test that verifies this works instead?

Reply:
Done!

Reply:
Thanks for adding the test! Looks like this is still broken for codegen tests though, can you add a test for that case too?

Reply:
Fixed `codegen` and `asm` test suites. I couldn't add a test though, because both suites expect the compilation to pass, and `// build-fail` doesn't work in non-UI suites.